src/main/java/org/spongepowered/api/service/ServiceManager.java

```diff
@@ -48,14 +50,14 @@
      * <p>Services should only be registered during initialization. If services
      * are registered later, then they may not be utilized.</p>
      *
-     * @param plugin The instance of a plugin
+     * @param cause The cause of the provider change.
      * @param service The service
      * @param provider The implementation
      * @param <T> The type of service
      * @throws IllegalArgumentException Thrown if {@code plugin} is not a
      *     plugin instance
      */
-    <T> void setProvider(Object plugin, Class<T> service, T provider);
+    <T> void setProvider(Cause cause, Class<T> service, T provider);
```

Comment:
I agree - I don't think a cause makes sense here. What would be in the cause, other than a plugin container?